### DISCUSSION TITLE
Ordered multiple quick picks

### DISCUSSION THREAD

**Discussion by PEZ** [microsoft/vscode-discussions#567] [Extension Development QnA] [answered]
Does anyone know of a way to let the user choose multiple options from a list, and also an order between them? Like if the list is

* foo
* bar
* baz
* gaz

And the user can select, e.g.

1. baz
2. bar

Doesn't need to be a quickpick, I just need to know the items selected and in which order the user wants them applied.

**Comment by charisk** [comment]
You can use a [multi-select quick-pick](https://code.visualstudio.com/api/ux-guidelines/quick-picks#multiple-selections).

To enable that you can use the `canPickMany` flag in the [QuickPickOptions](https://code.visualstudio.com/api/references/vscode-api#QuickPickOptions). 

```
window.showQuickPick(items, { canPickMany: true });
```

I believe the options show up in the order they appear in the `items` array.

**Reply by PEZ** [reply]
That's what I am using. I want to let the user specify an order.

**Reply by isidorn** [reply, marked as answer]
Currently not supported.

**Comment by vaclavHala** [comment]
@PEZ  If wizard-like multistep flow would be alright for you, you can do something along these lines

``` ts
const items = ['foo', 'bar', 'baz']
new Promise<string[]>(async (resolve, reject) => {
	const acc: string[] = []
	while(true){
		const next = await vscode.window.showQuickPick(items .concat( ["Done"]), {
			title: `Picked so far: ${acc}`,
			canPickMany: false
		})
		if(next === undefined){
			reject()
			return
		} else if (next === 'Done'){
			resolve(acc)
			return
		} else {
			acc.push(next)
		}
	}
}).then(
	acc => console.log("You selected: "+acc), 
	err => console.log("Selection canceled")
)
```
You could improve it further by e.g. also providing `Back` command which removes the last added item.
Also, if you need each item at most once just remove items already selected from subsequent quickpicks.
